FieldOpt/Optimization/optimizers/CMA_ES.h

```diff
@@ -0,0 +1,126 @@
+/******************************************************************************
+   Created by Brage on 18/06/19.
+   Copyright (C) 2019 Brage Strand Kristoffersen <brage_sk@hotmail.com>
+
+   This file is part of the FieldOpt project.
+
+   FieldOpt is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   FieldOpt is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with FieldOpt.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+#ifndef FIELDOPT_CMA_ES_H
+#define FIELDOPT_CMA_ES_H
+
+#include <boost/random.hpp>
+#include "optimizer.h"
+
+namespace Optimization {
+namespace Optimizers {
+
+
+/*!
+* @brief This class implements the "Covariance Matrix Adaption Evolutionary Strategy" (CMA-ES) algorithm.
+*
+* It samples the space using a multivariate normal distribution in each real-valued dimension. For each step the
+* variables in the population is perturbed to achieve a zero mean, based on the covariance matrix. For each step
+* the coverance matrix is based on the new distribution around the new mean.
+*
+* It approximates the
+* The implementation is based on the description found at:
+* http://delivery.acm.org/10.1145/3080000/3075986/p183-sakamoto.pdf?ip=129.241.230.160&id=3075986&acc=ACTIVE%20SERVICE&key=CDADA77FFDD8BE08%2E5386D6A7D247483C%2E4D4702B0C3E38B35%2E4D4702B0C3E38B35&__acm__=1560861322_554a2416a12a88617c76b0a945b1687c
```

Comment:
This link does not work now.
It's often best to just state the title, author, etc. and add the DOI.